### ISSUE TITLE
Cluster Becomes Unresponsive

### ISSUE THREAD

**Issue by SalvoLunar** [flynn/flynn#4395]
Hi there,

My cluster today became unresponsive, I suspect one of my nodes went temporarily down and that caused the issue.

To fix this I had to reinstall flynn-host with the --clean option and restore the cluster from a backup.

Is there any better way to fix this issue?

It keeps happening and I'm not exactly sure why.

Thanks,
Salvo

**Comment by SalvoLunar** [comment]
I'm getting this error when checking the status of flynn-host

May 26 19:01:28 scw-023b06 flynn-host[2370]: 2018/05/26 19:01:28 rpc: client protocol error: read unix @->/.container-shared/rpc.sock: EOF

**Comment by SalvoLunar** [comment]
Hi, this happened again today, twice

I think flynn-host randomly restarts ( checking the server log there are no reboots)  and it's unable to connect to the leader.

I've solved it by running 
`sudo systemctl stop flynn-host`
on all my nodes
and then 
`flynn-host fix --min-host 3`
on the leader.

Obviously I cannot keep doing this manually. 
Does anyone here have any suggestion on how to fix this?

Thanks,
Salvo

**Comment by thomasmodeneis** [comment]
Hi @SalvoLunar 
Are you using the postgres or your data is saved somewhere-else ?
I also experience this issue from time to time, and on my experience restoring the backup only works for few days.
The best is to completely remove the cluster and start new one with no backups. At least like this I can keep things online for longer, but it depends on the team activity and amount of deployments.

**Comment by SalvoLunar** [comment]
Hi @thomasmodeneis,

I'm using mysql, but I think I have fixed the issue, it was due to the limit in memory and cpu assigned to my apps.

When the load on my apps was going over the limit the process were crashing and somehow bringing the node down, the node was failing to reboot and the entire cluster was becoming unresponsive.

You can change the limits with the command below:

flynn -a myapp limit set web memory=2GB cpu=2000

Select the correct app using the -a parameter

Also, I've added a cron job onto my nodes that runs the garbage collector command ( flynn-host volume gc) every hour.

If you app is not using docker you can also stop the dockerreceive process. 
Same for redis ( if you are not using it just turn it off)

Hope that helps.

**Comment by SalvoLunar** [comment]
This morning suddenly the router stopped working.
The router service seems to be running but I've got this error from the logs:

`t=2018-06-11T07:47:47+0000 lvl=eror msg="unretriable request error" service=fourstar-production-web request_id=6722b70d-e1ca-4109-a4a3-7f5ac2aeccd5 client_addr=162.158.155.240:26109 host=fourstarpizza.ie path=/ method=GET status=503 job.id=scw74eed6-3eb5a130-b04e-4d51-a2d9-925c36d3665b addr=100.100.34.40:8080 err="context canceled" attempt=0`

trying to get the log info the process gets stuck at:

```
INFO[06-11|07:52:09] uploading logs and debug information to a private, anonymous gist 
INFO[06-11|07:52:09] this may take a while depending on the size of your logs 
INFO[06-11|07:52:09] getting flynn-host logs 
INFO[06-11|07:52:11] getting sirenia metadata 
INFO[06-11|07:52:11] getting scheduler state 
EROR[06-11|07:52:21] error getting scheduler state            err="Get http://100.100.83.8:5000/debug/state: net/http: request canceled (Client.Timeout exceeded while awaiting headers)"
**INFO[06-11|07:52:21] getting job logs** 
```
Can anyone give me any suggestion of how to fix this?

**Comment by SalvoLunar** [comment]
Hi there,

I think I've found a bug.

So if the any of servers in my cluster restarts for any reason my sites go down.

What I usually I've found is that the DB processes are unable to restore and even if I restart them they are running with the read-only option.

I've tried numerous times to run `flynn-host fix --min-hosts 3` which seems to fix the cluster but in reality it does nothing because if I check the mariadb logs I can see an infinite loops of "waitForUpstream" calls that don't resolve and the DB is still running with the --read-only option.

So, until now I coulnd't find a way to fix this without restoring everything from a backup.

Today I discovered that I can get the DB to work again correctly if I pass the --peer-ips option to the fix command.

`flynn-host fix --min-hosts=3 --peer-ips=<MY IPS>
`
So I believe there is a bug in the flynn-host fix command.

Hope that helps solving it.

Cheers,
Salvo
